qiskit/providers/aer/backends/qasm_simulator.py

```diff
@@ -160,17 +196,36 @@ def _validate(self, qobj, backend_options, noise_model):
                     clifford = False
                 if no_measure and op.name == "measure":
                     no_measure = False
+                if ch_supported and op.name in unsupported_ch_instructions:
+                    ch_supported = False
             # Print warning if clbits but no measure
             if no_measure:
                 logger.warning('No measurements in circuit "%s": '
                                'count data will return all zeros.', name)
             # Check qubits for statevector simulation
-            if not clifford:
+            if not clifford and method != "ch":
                 n_qubits = experiment.config.n_qubits
                 max_qubits = self.configuration().n_qubits
                 if n_qubits > max_qubits:
                     system_memory = int(local_hardware_info()['memory'])
-                    raise AerError('Number of qubits ({}) '.format(n_qubits) +
-                                   'is greater than maximum ({}) '.format(max_qubits) +
-                                   'for "{}" (method=statevector) '.format(self.name()) +
-                                   'with {} GB system memory.'.format(system_memory))
+                    err_string = ('Number of qubits ({}) is greater than '
+                                 'maximum ({}) for "{}" (method=statevector) '
+                                 'with {} GB system memory')
+                    err_string = err_string.format(n_qubits, max_qubits,
+                                                   self.name(), system_memory)
+                    if method != "automatic":
+                        raise AerError(err_string + '.')
+                    else:
+                        if n_qubits > 63:
+                            raise AerError(err_string + ', and has too many ' +
+                                           'qubits to fall back to the ' + 
+                                           'CH simulator.')
+                        elif not ch_supported:
```

Comment:
No need for `elif`,  a simple `if` will be enough as the above `raise` will terminate the function